src/config/CompactSerializationConfig.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CompactSerializer} from '../serialization/compact/CompactSerializer';
+
+/**
+ * Compact serialization config for the client.
+ *
+ * This API is currently in BETA and can change at any time.
+ */
+export interface CompactSerializationConfig {
```

Comment:
Yes, java has it. In Node.js I think we don't need it because there is no zero config support. Users must give compact serializers. Adding a boolean flag seems unnecessary to me. 

So compact is enabled for a class if there is a serializer for it. 

In Node.js client, if a user does not specify any compact serializers, enabling compact serialization would have no effect.